setup.py

```diff
@@ -166,6 +166,7 @@
     "aiohttp",
     "elasticsearch>=7.17.12,<8.0.0",  # 8.0 asks users to provide hosts or cloud_id when instantiating ElasticSearch(); 7.9.1 has legacy numpy.float_ which was fixed in https://github.com/elastic/elasticsearch-py/pull/2551.
     "faiss-cpu>=1.8.0.post1",  # Pins numpy < 2
+    "h5py",  # FIXME: probably needs a lower bound
```

Comment:
```suggestion
    "h5py>=2.3",  # FIXME: probably needs a lower bound
```
Probably the most recent thing we need is the vlen/complex/compound, which [according to the docs was added in 2.3](https://docs.h5py.org/en/stable/special.html#:~:text=As%20of%20version%202.3%2C%20h5py%20fully%20supports%20HDF5%20enums%20and%20VL%20types.). Version as of writing is 3.14